temporian/implementation/numpy/operators/reindex.py

```diff
@@ -0,0 +1,153 @@
+            dst_idx_metadata[dst_idx_lvl]["block_length"] = this_block_length
+            dst_idx_metadata[dst_idx_lvl]["timestamps"] = [timestamps]
+            dst_idx_metadata[dst_idx_lvl]["features"] = [
+                drop_feats + src_event.data[src_idx_lvl]
+            ]
+
+    # allocate memory for destination sampling & event
+    # sampling
+    dst_samp_data = {
+        idx_lvl: np.empty(metadata["block_length"])
+        for idx_lvl, metadata in dst_idx_metadata.items()
+    }
+    # destination feature names
+    dst_feat_names = drop_feat_names + src_event.feature_names
+
+    # event
+    dst_event_data = {
+        idx_lvl: [
+            NumpyFeature(
+                name=dst_feat_name, data=np.empty(metadata["block_length"])
+            )
+            for dst_feat_name in dst_feat_names
+        ]
+        for idx_lvl, metadata in dst_idx_metadata.items()
+    }
+    # assign data to previously allocated memory
+    for dst_idx_lvl, metadata in dst_idx_metadata.items():
+        ptr = 0
+        for timestamps, feats in zip(
+            metadata["timestamps"], metadata["features"]
+        ):
+            # number of samples in this index level
+            this_block_length = len(timestamps)
+
+            # allocate sampling
+            dst_samp_data[dst_idx_lvl][
+                ptr : ptr + this_block_length
+            ] = timestamps
+
+            # allocate event
+            for i, feat in enumerate(feats):
+                dst_event_data[dst_idx_lvl][i].data[
+                    ptr : ptr + this_block_length
+                ] = feat.data
+
+            # increment pointer
+            ptr += this_block_length
+
+    # finally, sort according to timestamps
+    for dst_idx_lvl in dst_event_data.keys():
+        sorted_idxs = np.argsort(dst_samp_data[dst_idx_lvl], kind="mergesort")
+
+        # sampling
+        dst_samp_data[dst_idx_lvl] = dst_samp_data[dst_idx_lvl][sorted_idxs]
+
+        # event
+        for feature in dst_event_data[dst_idx_lvl]:
+            feature.data = feature.data[sorted_idxs]
+
+    # create & return output NumpyEvent w/ it's NumpySampling
+    return NumpyEvent(
+        dst_event_data, NumpySampling(src_idx_names, dst_samp_data)
+    )
+
+
+class ReIndexNumpyImplementation:
+    """Select a subset of features from an event."""
+
+    def __init__(self, op: ReIndex) -> None:
+        assert isinstance(op, ReIndex)
+        self._op = op
+
+    def __call__(self, event: NumpyEvent) -> Dict[str, NumpyEvent]:
+        # sampling index names
+        src_index_names = event.sampling.names
+        dst_index_names = self._op.attributes()["dst_index"]
+
+        # fist case - destination index is a subset of source index
+        if set(dst_index_names).issubset(set(src_index_names)):
+            output_event = _subset_impl(event, src_index_names, dst_index_names)
+
+        elif set(dst_index_names).issuperset(set(src_index_names)):
+            pass
```

Comment:
Nit: Raise NotImplementedError. It's easier to debug if it's a superset and you didn't notice.